apps/opentelemetry_experimental/src/otel_log_handler.erl

```diff
@@ -128,111 +173,274 @@ log(LogEvent, _Config=#{regname := Id}) ->
+          type     => worker,
+          modules  => [?MODULE]},
+    case supervisor:start_child(?SUP, ChildSpec) of
+        {ok, _Pid} ->
+            {ok, Config#{reg_name => RegName}};
+        {error, {Reason, Ch}} when is_tuple(Ch), element(1, Ch) == child ->
+            {error, Reason};
+        {error, _Reason}=Error ->
+            Error
+    end.
+
+handle_event_(State, {timeout, check_table_size}, check_table_size,
+              Data=#data{max_queue_size=MaxQueueSize,
+                         max_export_batch_size=MaxBatchSize,
+                         check_table_size_ms=CheckInterval,
+                         reg_name=RegName}) ->
+    Size = ets:info(?CURRENT_TABLE(RegName), size),
+    case Size >= MaxQueueSize of
+        true -> disable(RegName);
+        false -> enable(RegName)
+    end,
+    case Size >= MaxBatchSize of
+        true when State =:= exporting ->
+            {keep_state_and_data, [postpone]};
+        true ->
+            %% By no means it can ensure that max_export_batch_size is never exceeded.
+            %% However, it should work fine to trigger export as soon as the table
+            %% reaches the max size instead of waiting for the scheduled export timeout,
+            %% that can be a quite large value.
+            {next_state, exporting, Data, [?check_tab_timeout(CheckInterval)]};
+        false ->
+            {keep_state_and_data, [?check_tab_timeout(CheckInterval)]}
+    end;
+handle_event_(_State, internal, init_exporter, Data=#data{exporter=undefined,
+                                                     exporter_config=ExporterConfig,
+                                                     reg_name=RegName}) ->
+    Exporter = init_exporter(RegName, ExporterConfig),
+    {keep_state, Data#data{exporter=Exporter}};
+
+handle_event_(_State, {call, From}, {changing_config, NewConfig}, Data) ->
+    {keep_state, add_config_to_data(NewConfig, Data), [{reply, From, {ok, NewConfig}}]};
+handle_event_(_State, _, _, _) ->
+    keep_state_and_data.
+
+init_exporter(RegName, ExporterConfig) ->
     case otel_exporter:init(ExporterConfig) of
         Exporter when Exporter =/= undefined andalso Exporter =/= none ->
+            enable(RegName),
             Exporter;
         _ ->
+            %% exporter is undefined/none
+            %% disable the insertion of new log events and delete the current table
+            clear_table_and_disable(RegName),
             undefined
     end.
 
+next_table(CurrentTab, Tab1, Tab2) when CurrentTab =:= Tab1 ->
+    Tab2;
+next_table(_CurrentTab, Tab1, _Tab2) -> Tab1.
+
+maybe_export_other_table(CurrentTab, Data=#data{table_1=Tab1,
+                                                table_2=Tab2,
+                                                exporting_timeout_ms=ExportingTimeout,
+                                                scheduled_delay_ms=SendInterval}) ->
+    NextTab = next_table(CurrentTab, Tab1, Tab2),
+    case ets:info(NextTab, size) of
```

Comment:
this is quite expensive when the table has write concurrency set to true.
maybe this:

```
is_ets_table_empty(Tab) -
    ets:first(Tab) =:= '$end_of_table'.
```

Reply:
I think retrieving size is only expensive when `decentralized_counters` are enabled (or `write_concurrency` set to `auto`) isn't it? 
Anyway, this is a subject for some additional research. Currently, opentelemetry scope uses quite a static key: [instrumentation scope](https://github.com/SergeTupchiy/otel_bench/blob/main/results.md#traverse-results).
Meaning that all the writes are done to only one key (by default) and probably just a few if a user configures several scopes.. 
So, probably enabling write_concurrency doesn't really help when all inserts are done to the same key. 
I've already done some [benchmarks](https://github.com/SergeTupchiy/otel_bench/blob/main/results.md#insert-results) (set vs duplicate_bag insert), generally `set` looked faster ~5-10% (1000 concurrent processes each one inserting a record in a loop).

Reply:
I think using `ets:first(Tab) =:= '$end_of_table'.` is a great alternative, thanks for suggestion :raised_hands: